### DISCUSSION TITLE
Do not raise SIM115 when a file is closed immediately?

### DISCUSSION THREAD

**Discussion by harupy** [astral-sh/ruff#7907] [Q&A] [answered]
The following code violates `SIM115`:

```python
open("a.txt", "w").close()  # create or reset a.txt

# a.py:1:1: SIM115 Use context handler for opening files
```

I'm wondering if SIM115 should not be raised in this case. We could use the following as a workaround:

```python
with open("a.txt", "w"):
    pass
```

**Comment by charliermarsh** [comment, marked as answer]
Yeah, I think we should allow this.

**Reply by harupy** [reply]
https://github.com/MartinThoma/flake8-simplify/issues/174 is related.
